ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -93,24 +84,28 @@ public AzureBlobDataAccessor(CloudConfig cloudConfig, AzureCloudConfig azureClou
     uploadTimeout = Duration.ofMillis(cloudConfig.cloudUploadRequestTimeout);
     batchTimeout = Duration.ofMillis(cloudConfig.cloudBatchRequestTimeout);
 
-    StorageClientFactory storageClientFactory = Utils.getObj(azureCloudConfig.azureStorageClientFactoryClass);
-    storageClient = storageClientFactory.createBlobStorageClient(cloudConfig, azureCloudConfig);
-    blobBatchClient = new BlobBatchClientBuilder(storageClient).buildClient();
+    storageClient =
+        Utils.getObj(azureCloudConfig.azureStorageClientClass, azureCloudConfig, cloudConfig, blobLayoutStrategy);
   }
 
   /**
    * Test constructor
-   * @param storageClient the {@link BlobServiceClient} to use.
+   * @param blobServiceClient the {@link BlobServiceClient} to use.
    * @param blobBatchClient the {@link BlobBatchClient} to use.
    * @param clusterName the cluster name to use.
    * @param azureMetrics the {@link AzureMetrics} to use.
    */
-  AzureBlobDataAccessor(BlobServiceClient storageClient, BlobBatchClient blobBatchClient, String clusterName,
+  AzureBlobDataAccessor(BlobServiceClient blobServiceClient, BlobBatchClient blobBatchClient, String clusterName,
       AzureMetrics azureMetrics) {
-    this.storageClient = storageClient;
     this.blobLayoutStrategy = new AzureBlobLayoutStrategy(clusterName);
+    try {
+      this.storageClient =
+          Utils.getObj(AzureCloudConfig.DEFAULT_AZURE_STORAGE_CLIENT_CLASS, blobServiceClient, blobBatchClient,
```

Comment:
It's kind of unconventional to instantiate impl's instead of factories, but I'm not strongly opposed to it.